bin/ember-template-lint.js

```diff
@@ -58,35 +60,62 @@ function printErrors(errors) {
 }
 
 function lintFile(linter, filePath, moduleId) {
-  var source = fs.readFileSync(filePath, { encoding: 'utf8' });
+  var source;
+  try {
+    source = rw.readFileSync(filePath, { encoding: 'utf8' });
+  } catch (error) {
+    if (error.code === 'ENXIO') {
+      return [];
+    } else {
+      throw error;
+    }
+  }
   return linter.verify({ source: source, moduleId: moduleId });
 }
 
-function getRelativeFilePaths() {
-  var fileArgs = process.argv.slice(2).filter(arg => arg.slice(0, 2) !== '--');
-
-  var relativeFilePaths = fileArgs
-    .reduce((filePaths, fileArg) => {
-      return filePaths.concat(
-        globby.sync(fileArg, {
+function expandFileGlobs(fileArgs) {
+  return fileArgs.reduce((filePaths, fileArg) => {
+    var files;
+    if (['-', STDIN].includes(fileArg)) {
+      if (filePaths.includes(STDIN)) {
+        return filePaths;
+      }
+      files = [STDIN];
+    } else {
+      files = globby
+        .sync(fileArg, {
           ignore: ['**/dist/**', '**/tmp/**', '**/node_modules/**'],
           gitignore: true,
         })
-      );
-    }, [])
-    .filter(filePath => filePath.slice(-4) === '.hbs');
+        .filter(filePath => filePath.slice(-4) === '.hbs');
+    }
+    return filePaths.concat(files);
+  }, []);
+}
+
+function getRelativeFilePaths() {
+  var fileArgs = process.argv.slice(2).filter(arg => arg.slice(0, 2) !== '--');
```

Comment:
I'm a tad confused how we are avoiding reading the value of `--filename` as one of the glob paths here. AFAICT with the following invocation:

```
ember-template-lint --filename app/templates/application.hbs < /some/other/path/app/templates/application.hbs
```

This line of code would result in:

```js
var fileArgs = ['app/templates/application.hbs'];
```

Which I believe is incorrect.

I think in order to combat these sorts of issues, I'd like to see us move to a smallish `parseArgv` helper function (but would prefer to avoid pulling in `yargs` or `commander`). 

_Something_ like:

```js
function parseArgv(argv) {
  let options = {
    positional: [],
    named: {},
  };

  for (let i = 0; i < argv.length; i++) {
    let arg = argv[i];
    
    if (arg.startsWith('--')) {
      let value = argv[i + 1];
	  let optionName = arg.substring(2);

      options.named[optionName] = value;

      i++;
    }

    options.positional.push(arg);
  }
}

const options = parseArgv(process.argv.slice(2));
```

Then throughout the rest of the script we can refer to `options.positional` as the `fileArgs` and our passed flags are in `options.named`.  

I _think_ this will both fix the bug I mentioned above and simplify the code overall. What do you think?

Reply:
Seems legit to me. I was attempting to copy previous precedence from `--config-path`. An argv parser is much nicer.

Reply:
Ya, I definitely saw that you were following the existing style. It just felt like “time” to move to a better parsing situation.